core/src/main/java/jenkins/model/Detail.java

```diff
@@ -0,0 +1,74 @@
+package jenkins.model;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import hudson.model.Actionable;
+import hudson.model.ModelObject;
+import hudson.model.Run;
+import org.jenkins.ui.icon.IconSpec;
+
+/**
+ * {@link Detail} represents a piece of information about a {@link Run}.
+ * Such information could include:
+ * <ul>
+ *  <li>the date and time the run started</li>
+ *  <li>the amount of time the run took to complete</li>
+ *  <li>SCM information for the build</li>
+ *  <li>who kicked the build off</li>
+ * </ul>
+ * @since TODO
+ */
+public abstract class Detail implements ModelObject, IconSpec {
+
+    private final Actionable object;
+
+    public Detail(Actionable object) {
+        this.object = object;
+    }
+
+    public Actionable getObject() {
+        return object;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public @Nullable String getIconClassName() {
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public @Nullable String getDisplayName() {
+        return null;
+    }
+
+    /**
+     * Optional URL for the {@link Detail}
+     */
+    public @Nullable String getUrl() {
+        return null;
+    }
+
+    /**
+     * Returns true if this detail is applicable to the given Actionable object
+     */
+    public boolean isApplicable() {
```

Comment:
For `Action` this would be the job of the factory. Why create an object just to discard it?

Reply:
That's fair, I was using `isApplicable` to encapsulate show/hide logic of the detail inside of the `Detail`, so that its factory didn't need to worry about that.